src/psygnal/_signal.py

```diff
@@ -1242,16 +1242,26 @@ def _is_subclass(left: Type[Any], right: type) -> bool:
     return issubclass(left, right)
 
 
+_PARTIAL_CACHE: Dict[int, Tuple[weakref.ref, str, Callable]] = {}
```

Comment:
I have significant worries about memory leaking here. There should be at least some cleaning steps executed from time to time.

Reply:
what are you thinking would take up much memory here?  the key is an integer, the value is a string of a weakref, string, and callable... are you worried about the args/kwargs closure on the callable?

Reply:
I added a _prune_cache method that will get called either when a partial is disconnected, or when an emit loop finds a dead weakfref

Reply:
My main worries are connected with scenarios like what happens in napari EventedModels. There are multiple steps of creating objects and destroying them shortly after. 

If I understand correctly if user often creates a short living object that will be removed later and them will be used connection, then this dictionary may grows?

Reply:
> If I understand correctly if user often creates a short living object that will be removed later and them will be used connection, then this dictionary may grows?

before my last commit, yes, there was the possibility for it to grow... but my main point above is that this dictionary is _miniscule_... it's using integers and strings as values for the most part... not the original object.

however, after my last commit https://github.com/tlambert03/psygnal/pull/134/commits/221ad11db92503e0e08f26ff1e5222fd64b4e6b7

when a slot is disconnected (either explicitly using `disconnect` or implicitly when the `_run_emit_loop` method encounters a dead weakref) then `_prune_partial_cache()` will be called that trims the dict